orchagent/dash/dashaclorch.cpp

```diff
@@ -0,0 +1,873 @@
+    }
+
+}
+
+template<>
+inline void lexical_convert(const string &buffer, sai_ip_addr_family_t &ip_version)
+{
+    SWSS_LOG_ENTER();
+
+    if (buffer == "ipv4")
+    {
+        ip_version = SAI_IP_ADDR_FAMILY_IPV4;
+    }
+    else if (buffer == "ipv6")
+    {
+        ip_version = SAI_IP_ADDR_FAMILY_IPV6;
+    }
+    else
+    {
+        SWSS_LOG_ERROR("Invalid ip version : %s", buffer.c_str());
+        throw invalid_argument("Invalid ip version");
+    }
+}
+
+template<>
+inline void lexical_convert(const string &buffer, DashAclRuleEntry::Action &action)
+{
+    SWSS_LOG_ENTER();
+
+    if (buffer == "allow")
+    {
+        action = DashAclRuleEntry::Action::ALLOW;
+    }
+    else if (buffer == "deny")
+    {
+        action = DashAclRuleEntry::Action::DENY;
+    }
+    else
+    {
+        SWSS_LOG_ERROR("Invalid action : %s", buffer.c_str());
+        throw invalid_argument("Invalid action");
+    }
+}
+
+template<>
+inline void lexical_convert(const string &buffer, bool &terminating)
+{
+    SWSS_LOG_ENTER();
+
+    if (buffer == "true" || buffer == "True" || buffer == "1")
+    {
+        terminating = true;
+    }
+    else if (buffer == "false" || buffer == "False" || buffer == "0")
+    {
+        terminating = false;
+    }
+    else
+    {
+        SWSS_LOG_ERROR("Invalid terminating : %s", buffer.c_str());
+        throw std::invalid_argument("Invalid terminating");
+    }
+}
+
+template<>
+inline void lexical_convert(const string &buffer, vector<uint8_t> &protocols)
+{
+    SWSS_LOG_ENTER();
+
+    auto tokens = tokenize(buffer, ',');
+    protocols.clear();
+    protocols.reserve(tokens.size());
+    for (auto &token : tokens)
+    {
+        uint32_t protocol;
+        lexical_convert(token, protocol);
+        protocols.push_back(static_cast<uint8_t>(protocol));
+    }
+}
+
+template<>
+inline void lexical_convert(const string &buffer, vector<sai_ip_address_t> &ips)
+{
+    SWSS_LOG_ENTER();
+
+    auto tokens = tokenize(buffer, ',');
+    ips.clear();
+    ips.reserve(tokens.size());
+    for (auto &token : tokens)
+    {
+        sai_ip_address_t ip;
+        IpAddress ipAddress(token);
+        swss::copy(ip, ipAddress);
+        ips.push_back(ip);
+    }
+}
+
+template<>
+inline void lexical_convert(const string &buffer, std::vector<sai_u16_range_t> &ports)
+{
+    SWSS_LOG_ENTER();
+
+    auto tokens = tokenize(buffer, ',');
+    ports.clear();
+    ports.reserve(tokens.size());
+    for (auto &token : tokens)
+    {
+        sai_u16_range_t port;
+        if (token.find('-') == std::string::npos)
+        {
+            // Only one port
+            lexical_convert(token, port.min);
+            port.max = port.min;
+        }
+        else if (!extractVariables(token, '-', port.min, port.max))
+        {
+            // Range
+            SWSS_LOG_ERROR("Invalid port range : %s", token.c_str());
+            throw invalid_argument("Invalid port range");
+        }
+        ports.push_back(port);
+    }
+}
+
+}
```

Comment:
is this brace needed? seems to be extra

Reply:
It's needed for line: https://github.com/sonic-net/sonic-swss/blob/957e9288c98a4d026edb65e3c33c8ca1a2bd5582/orchagent/dash/dashaclorch.cpp#L61